traitsui/qt4/testing/traitsui_test_assistant.py

```diff
@@ -0,0 +1,142 @@
+#  Copyright (c) 2019, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+
+from contextlib import contextmanager
+
+from pyface.ui.qt4.util.gui_test_assistant import GuiTestAssistant
+from traits.trait_base import xgetattr
+
+
+class TraitsUITestAssistant(GuiTestAssistant):
+    """ A Test Assistant for testing TraitsUI views.
+
+    For most efficient usage, this class expects the setUp to provide an
+    'object' attribute containing the main object being viewed.
+
+    Note
+    ----
+
+    This API is experimental and may change before the next feature release.
+    """
+
+    @contextmanager
+    def create_ui(self, object=None, **kwargs):
+        """ Create and dispose of a TraitsUI UI
+
+        This is a context manager which calls edit_traits on an object in
+        when entered and calls dispose when exiting, running the event
+        loop until the underlying toolkit object is destroyed.
+
+        Parameters
+        ----------
+        object : HasTraits instance or None
+            The object to call edit_traits on.  If None, then the method
+            will attempt to use self.object, if it exists.
+        **kwargs : keyword arguments
+            Additional keyword arguments will be passed through to
+            edit_traits.
```

Comment:
Drive-by comment: As a user of this method, I would not mind having to pass the entire dictionary of keyword arguments as one argument, e.g.  `create_ui(object=some_instance, edit_kwargs=dict(view=my_view))`  and then `object.edit_traits(**edit_kwargs)`. Then there will be room for future extension to the signature of `create_ui`.